### PR TITLE
Feature/191a line number

### LINKED ISSUE #192: Upgrade to Exasol 8.0
Links to https://github.com/exasol/roadmap/issues/76 

When Exasol 8.0 is available on docker hub, we need to upgrade exasol-testcontainers to use the new release in order to verify everything works with the new release.

Branch for the upgrade already exists: https://github.com/exasol/exasol-testcontainers/tree/feature/%23191-support-exasol-8

## Acceptance criteria
* [ ] All tests are green

### PR DESCRIPTION
**Pull request by ckunki**
fixes #192 

Additionally updated bucketfs-java to version 2.4.0 with generic state-based monitoring for SyncAwareBuckets.  State-based monitors support different strategies for filtering events from the BucketFS logfile. Besides evaluating the timestamps of the log entries as in the past another strategy is to evaluate their line number and to accept only log entries with higher line number than then log files's size was before triggering the monitored operation.

### CODE REVIEW

File: src/main/java/com/exasol/bucketfs/testcontainers/TestcontainerBucketFactory.java [hunk 1]
```diff
@@ -51,16 +45,84 @@ public synchronized Bucket getBucket(final String serviceName, final String buck
...
+        });
         return this.buckets.get(bucketPath);
     }
+
+    private int mapPort(final int internalPort) {
+        return this.portMappings.get(internalPort);
+    }
+
+    /**
+     * Builder to create new instances of {@link TestcontainerBucketFactory}
```
Comment (kaklakariada) [inline src/main/java/com/exasol/bucketfs/testcontainers/TestcontainerBucketFactory.java:71]:
```suggestion
     * Builder to create new instances of {@link TestcontainerBucketFactory}.
```

File: src/main/java/com/exasol/clusterlogs/LogEntryPatternVerifier.java [hunk 2]
```diff
@@ -6,6 +6,21 @@
  */
 interface LogEntryPatternVerifier {
 
+    /**
+     * A {@link LogEntryPatternVerifier} which simulates to find any log message.
```
Comment (kaklakariada) [inline src/main/java/com/exasol/clusterlogs/LogEntryPatternVerifier.java:10]:
```suggestion
     * A {@link LogEntryPatternVerifier} which accepts any log message.
```
❓ 
Why do we need this in production code?

File: src/main/java/com/exasol/clusterlogs/LogPatternDetector.java [hunk 3]
```diff
@@ -120,4 +125,78 @@ public String getActualLog() {
...
+            throw new IllegalArgumentException("Unsupported class " + state.getClass() + " of state");
+        }
+
+        /**
+         * According to bucketfs developers:
+         * <ul>
+         * <li>bucketfsd does not use timezone setting from {@code /exa/etc/EXAConf} but hard-coded python
+         * {@code time.gmtime}</li>
+         * <li>there are discussions to unify all logs to UTC</li>
+         * </ul>
```
Comment (kaklakariada) [inline src/main/java/com/exasol/clusterlogs/LogPatternDetector.java:168]:
```suggestion
         * Bucketfsd in Exasol 8 does not use the timezone setting from {@code /exa/etc/EXAConf} but hard-coded python
         * {@code time.gmtime}.
```

File: src/main/java/com/exasol/containers/ExasolContainer.java [hunk 4]
```diff
@@ -523,17 +545,29 @@ protected void waitForBucketFs() {
...
+            return;
+        }
+
+        this.status.setServiceStatus(UDF, NOT_READY);
+        new UdfContainerWaitStrategy(this.detectorFactory).waitUntilReady(this);
+        this.status.setServiceStatus(UDF, READY);
+    }
+
+    /**
+     * @return if language containers are extracted by default. In Exasol database version 8 and above this is the case
```
Comment (kaklakariada) [inline src/main/java/com/exasol/containers/ExasolContainer.java:566]:
```suggestion
     * @return {@code true} if language containers are extracted by default. In Exasol database version 8 and above this is the case
```

File: src/test/java/com/exasol/bucketfs/testcontainers/BucketIT.java [hunk 5]
```diff
@@ -68,13 +58,38 @@ void testUploadFile(@TempDir final Path tempDir)
...
         final RandomFileGenerator generator = new RandomFileGenerator();
         final Path path = tempDir.resolve(Path.of(fileName));
         generator.createRandomFile(path, sizeInKiB);
         return path;
     }
 
+    // sample log messages:
+    // [I 220812 11:26:06 bucketfsd:228] rsync for id (('bfsdefault', 'default', 'dir1/file.txt')) is done
+    // [I 220812 11:10:21 bucketfsd:228] rsync for id (('bfsdefault', 'default', 'dir4/file.txt')) is done
+    // [I 220812 10:57:23 bucketfsd:228] rsync for id (('bfsdefault', 'default', 'dir5/sub5/file.txt')) is done
```
Comment (kaklakariada) [inline src/test/java/com/exasol/bucketfs/testcontainers/BucketIT.java:92]:
```suggestion
```
Is this needed here?

File: src/test/java/com/exasol/clusterlogs/LogPatternDetectorFactoryTest.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Mock
+    private ExasolContainer<? extends ExasolContainer<?>> containerMock;
+    private final LogPatternDetectorFactory factory = new LogPatternDetectorFactory(this.containerMock);
+
+    @Test
+    void unsupportedState() {
+        final State state = Mockito.mock(State.class);
+        assertThrows(IllegalArgumentException.class,
+                () -> this.factory.createLogPatternDetector("logpath", "logname", "pattern", state));
```
Comment (kaklakariada) [inline src/test/java/com/exasol/clusterlogs/LogPatternDetectorFactoryTest.java:28]: 👍

File: src/test/java/com/exasol/clusterlogs/MappedClusterLogsIT.java [hunk 7]
```diff
@@ -34,6 +34,7 @@ private static Path createTempDir() {
     @Test
     // [itest->dsn~mapping-the-log-directory-to-the-host~1]
     void testMapClusterLogs() throws InterruptedException, IOException {
-        assertThat(TEMP_DIR.resolve("syslog").toFile().exists(), equalTo(true));
+        final File syslogFile = TEMP_DIR.resolve("syslog").toFile();
+        assertThat("File " + syslogFile + " exists", syslogFile.exists(), equalTo(true));
```
Comment (kaklakariada) [inline src/test/java/com/exasol/clusterlogs/MappedClusterLogsIT.java:38]: 👍

File: doc/changes/changes_6.2.0.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
+# Test containers for Exasol on Docker 6.2.0, released 2022-08-18
+
+Code name: Upgrade to Exasol 8.0.
+
+## Summary
+
+Upgraded to Exasol 8.0:
+* small changes in format of entries in BucketFS log
+* assuming time zone UTC for entries in BucketFS log
+* disabled features discontinued in version 8, e.g. ExaOperation
```
Comment (kaklakariada) [inline doc/changes/changes_6.2.0.md:10]:
```suggestion
* disabled tests for Exasol 8 for discontinued features, e.g. ExaOperation
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml [hunk 9]
```diff
@@ -33,16 +39,38 @@ jobs:
         run: echo 'testcontainers.reuse.enable=true' > "$HOME/.testcontainers.properties"
       - name: Run tests and build with Maven
         run: |
+          echo
+          echo "--------------------------------------------------------------"
+          echo " Running matrix test with docker_db_version ${{ matrix.docker_db_version }} "
+          echo "--------------------------------------------------------------"
+          echo
+          echo "Support information archives are written to $SUPPORT_DIR"
```
[the remark this answers is not in the document]
Comment (ckunki) [reply .github/workflows/ci-build.yml:47]: in line 60:
```
SUPPORT_DIR: ${{ github.workspace }}/target/support
```

File: src/main/java/com/exasol/bucketfs/monitor/FilesizeState.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
+package com.exasol.bucketfs.monitor;
+
+import com.exasol.bucketfs.monitor.BucketFsMonitor.State;
+
+/**
+ * Only accept other states with higher line number.
+ */
+public class FilesizeState implements BucketFsMonitor.State {
+
+    private final Long lineNumber;
```
[the remark this answers is not in the document]
Comment (ckunki) [reply src/main/java/com/exasol/bucketfs/monitor/FilesizeState.java:10]: I think there was a reason, but maybe the reason has gone.

File: src/test/java/com/exasol/clusterlogs/LogPatternDetectorFactoryTest.java (hunk 6 above)
Comment (ckunki) [reply src/test/java/com/exasol/clusterlogs/LogPatternDetectorFactoryTest.java:28]: Thanks - I can't even remember to have created this test 😃

File: src/main/java/com/exasol/bucketfs/monitor/FilesizeState.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (ckunki) [reply src/main/java/com/exasol/bucketfs/monitor/FilesizeState.java:10]: using long, now

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**